### DISCUSSION TITLE
How to sort M2A entries (Rest API)

### DISCUSSION THREAD

**Discussion by scouech** [directus/directus#4587] [Archive]
_Context_

I have a collection called `folders` with a M2A relationship field called `items`
This field has a sort field called `position`
When I get an entry from my `folders` collection, I see my list of `items`, each have a `position` 
(i.e. position: 2, position:1, position: 3)

_Question_

I was wondering if it's possible to sort the items so that they appear in the right position when calling `/folders/:id` ?

Thanks!

**Comment by rijkvanzanten** [comment]
> This field has a sort field called position

I'm assuming that the `position` field exists on the junction table between `folders` and `items` correct? So you'd have `folders` -> `folders_items` -> (one of "many A", "many B" etc)

You can use the `deep` query parameter to put any of the "regular" query parameters on the nested data set, so you should be able to use

```
GET /items/folders
  ?deep[items][_sort]=position
```

**Reply by scouech** [reply]
hey @rijkvanzanten ! That's exactly it! Amazing!!! Thank you so much, that's saving me a lot of time 👍
